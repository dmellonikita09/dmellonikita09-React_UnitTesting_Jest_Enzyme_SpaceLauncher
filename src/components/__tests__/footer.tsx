import { shallow } from 'enzyme';
import React from 'react';
import Footer from '../footer';

describe('Footer', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Footer />);
    const home = wrapper.find('[to="/"]');
    expect(home).toHaveLength(1);
  });

  it('renders without error', () => {
    const wrapper = shallow(<Footer />);
    const cart = wrapper.find('[to="/cart"]');
    expect(cart).toHaveLength(1);
  });

  it('renders without error', () => {
    const wrapper = shallow(<Footer />);
    const profile = wrapper.find('[to="/profile"]');
    expect(profile).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
