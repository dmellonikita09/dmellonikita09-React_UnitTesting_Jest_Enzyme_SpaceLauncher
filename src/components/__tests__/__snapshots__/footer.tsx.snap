// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<Styled(footer)>
  <Styled(div)>
    <Styled(Link)
      to="/"
    >
      <ForwardRef(SvgHome) />
      Home
    </Styled(Link)>
    <Styled(Link)
      to="/cart"
    >
      <ForwardRef(SvgCart) />
      Cart
    </Styled(Link)>
    <Styled(Link)
      to="/profile"
    >
      <ForwardRef(SvgProfile) />
      Profile
    </Styled(Link)>
    <LogoutButton />
  </Styled(div)>
</Styled(footer)>
`;
